src/nvim/msgpack_rpc/server.c

```diff
@@ -52,33 +54,40 @@ static PMap(cstr_t) *servers = NULL;
 /// Initializes the module
 bool server_init(void)
```

Comment:
That was from one of my old PRs (https://github.com/splinterofchaos/neovim/commit/792882e9e415e7dfb26e0778d760003a7d4747e1). It used to return `void`, and maybe it should return an `int` like `server_start()`, but as you note, we don't do anything about failure (yet), the idea was just that we might want to at some point.